src/services/refactors/ConvertToGetterAndSetter.ts

```diff
@@ -0,0 +1,151 @@
+        if (!fieldInfo) return undefined;
+
+        const changeTracker = textChanges.ChangeTracker.fromContext(context);
+        const newLineCharacter = getNewLineOrDefaultFromHost(context.host, context.formatContext.options);
+
+        const { fieldName, accessorName, propertyDeclaration, needUpdateName, hasModifiers, needUpdateModifiers } = fieldInfo;
+        const accessorModifiers = hasModifiers ? createNodeArray([createToken(SyntaxKind.PublicKeyword)]) : undefined;
+
+        const getAccessor = generateGetAccessor(propertyDeclaration, fieldName, accessorName, accessorModifiers);
+        const setAccessor = generateSetAccessor(propertyDeclaration, fieldName, accessorName, accessorModifiers);
+
+        const modifiers = needUpdateModifiers ? createNodeArray([createToken(SyntaxKind.PrivateKeyword)]) : propertyDeclaration.modifiers;
+        if (needUpdateName || needUpdateModifiers) {
+            changeTracker.replaceNode(file, propertyDeclaration, updateOriginPropertyDeclaration(propertyDeclaration, fieldName, modifiers), {
+                suffix: newLineCharacter
+            });
+        }
+
+        changeTracker.insertNodeAfter(file, propertyDeclaration, getAccessor);
+        changeTracker.insertNodeAfter(file, propertyDeclaration, setAccessor);
+
+        return {
+            edits: changeTracker.getChanges(),
+            renameFilename: undefined,
+            renameLocation: undefined,
+        };
+    }
+
+    interface Info { originName: string; fieldName: string; accessorName: string; propertyDeclaration: PropertyDeclaration; needUpdateName: boolean; hasModifiers: boolean; needUpdateModifiers: boolean; }
+    function getConvertibleFieldAtPosition(file: SourceFile, startPosition: number): Info | undefined {
+        const node = getTokenAtPosition(file, startPosition, /*includeJsDocComment*/ false);
+        const propertyDeclaration = findAncestor(node.parent, isPropertyDeclaration);
+
+        if (!(propertyDeclaration && propertyDeclaration.name.kind === SyntaxKind.Identifier &&
+            (getModifierFlags(propertyDeclaration) | ModifierFlags.AccessibilityModifier) === ModifierFlags.AccessibilityModifier)) return undefined;
+
+        const containerClass = getContainingClass(propertyDeclaration);
+        if (!containerClass) return undefined;
+
+        const members = getMembersOfDeclaration(containerClass);
+        if (!members) return undefined;
+
+        const needUpdateName = propertyDeclaration.name.text.charCodeAt(0) !== CharacterCodes._;
+
+        const accessorName = needUpdateName ? propertyDeclaration.name.text : propertyDeclaration.name.text.substring(1);
+        const fieldName = `_${accessorName}`;
+
+        if (find(members, member => needUpdateName ? member.name.getText() === fieldName : member.name.getText() === accessorName)) return undefined;
+
+        const hasModifiers = !!find(members, member => !!member.modifiers);

```

Comment:
Hmm, I think I initially meant whether a field had a `public` modifier explicitly

Reply:
what if the field is Protected